source/grid/grid_tools.cc

```diff
@@ -2989,8 +2989,19 @@ namespace GridTools
               }
             else for (unsigned int i=0; i<patch.size(); ++i)
                 {
+                  // Even though vertices may not be exactly the same, the
+                  // appropriate cells will match since == for TriAccessors
+                  // checks only cell level and index.
                   if (patch_to_global_tria_map_tmp[active_tria_cell]==patch[i])
                     {
+                      // adjust the cell vertices of the local_triangulation to
+                      // match cell vertices of the global triangulation
+                      for (unsigned int v=0; v<GeometryInfo<Container::dimension>::vertices_per_cell; ++v)
+                        active_tria_cell->vertex(v) = patch[i]->vertex(v);
+
+                      Assert(active_tria_cell->center().distance( patch_to_global_tria_map_tmp[active_tria_cell]->center())<=1e-15*active_tria_cell->diameter(),
```

Comment:
Please break the condition in a way that it (roughly) fits into 80 character length lines.